src/main/java/hudson/matrix/LabelAxis.java

```diff
@@ -86,16 +85,13 @@ public boolean isInstantiable() {
             return !j.getNodes().isEmpty() || !j.clouds.isEmpty();
         }
 
-
-        public String buildLabelCheckBox(LabelAtom la, LabelAxis instance) {
+        public String buildLabelCheckBox(LabelAtom la) {
             final String escapedName = jsStringEscape(htmlAttributeEscape(la.getName()));
             final String escapedDescription = jsStringEscape(StringUtils.isEmpty(la.getDescription()) ? "" :
                     htmlAttributeEscape(la.getDescription()));
-            return new StringBuilder("\"").append(jsStringEscape("<input type='checkbox' name='values' json='")).append(escapedName).append(jsStringEscape("' "))
-                    .append("\"").append(String.format("+has(\"%s\")+", escapedName)).append("\"")
-                    .append(jsStringEscape("/><label class='attach-previous'>")).append(escapedName).append(" (").append(escapedDescription).append(")</label>\"")
-                    .toString();
-            // '${h.jsStringEscape('<input type="checkbox" name="values" json="'+h.htmlAttributeEscape(l.name)+'" ')}'+has("${h.jsStringEscape(l.name)}")+'${h.jsStringEscape('/><label class="attach-previous">'+l.name+' ('+l.description+')</label>')}'
+            return "<input type='checkbox' name='values' json='" +
```

Comment:
Can revert this if there are objections, this was a refactoring suggested by IDE.

Reply:
FTR That code was a correction of a previous vulnerability, not sure the current version does not reintroduce issues.

With `new YAHOO.widget.HTMLNode` I am pretty sure it interprets the values as HTML. Have you tried to reproduce the previous vulnerability or to inject some payloads to understand the behavior? :)

Reply:
I think lines 89 and 90 are related to the vulnerability fix, unless you mean a different one that I'm not aware of.
I've checked with simple payloads I could think of, and content was escaped properly. I'll give it another shot too to make sure.

Reply:
I've re-checked, it seems fine to me. Tried couple different payloads, couldn't break it. User provided content is escaped. I'd appreciate someone else trying to break it though.